pcsx2-qt/QtHost.cpp

```diff
@@ -1422,6 +1466,35 @@ void Host::CommitBaseSettingChanges()
 		return;
 	}
 
+	auto lock = Host::GetSettingsLock();
+	if (s_settings_save_timer)
+		return;
+
+	s_settings_save_timer = new QTimer;
+	s_settings_save_timer->connect(s_settings_save_timer, &QTimer::timeout, &QtHost::SaveSecretsSettings);
+	s_settings_save_timer->setSingleShot(true);
+	s_settings_save_timer->start(SETTINGS_SAVE_DELAY);
+
```

Comment:
This is reusing the same timer variable for multiple different purposes, it looks like only the first one will actually work at a time. Also, it looks like they're the wrong way round (Host::CommitBaseSettingChanges connects QtHost::SaveSecretsSettings vice versa.)